### DISCUSSION TITLE
Unable to load class [org.hsqldb.jdbcDriver] problemas al usar HSQLDB

### DISCUSSION THREAD

**Discussion by jcastagno99** [dds-jv/foro#324] [Consultas de herramientas] [answered]
Buenas, estoy teniendo un problema para correr HSQLDB, con la conexión a mysql me funciona todo bien, pero cuando quiero conectar la base de datos en memoria me tira este error: 
![image](https://user-images.githubusercontent.com/49038828/139748572-b23c8e2d-ccb1-41ea-88aa-0d9f4363822d.png)

Basicamente estoy buscando replicar el ejemplo de spark en el tp, cree la clase Servidor y la clase Bootstrap

![image](https://user-images.githubusercontent.com/49038828/139748695-4051df99-53be-4ee2-8e42-f9d608dc7b61.png)

Bootstrap crea unas asociaciones y les dice al repo que las guarde

![image](https://user-images.githubusercontent.com/49038828/139748720-22f01f34-c49c-4ce8-98de-f0f01d581ca5.png)

En el repo llama al metodo agregarAsociacion

![image](https://user-images.githubusercontent.com/49038828/139748802-8b41e64d-a85c-40a4-887f-c1112aec6759.png)

En el pom tengo esta dependencia: 

![image](https://user-images.githubusercontent.com/49038828/139749259-2312b2fa-91b7-489c-8fb0-a69e5c0972c0.png)

Versiones mas actuales no me las toma, no se si es por la versión de maven.

En el persistance que esta en main/resources/META-INF/

![image](https://user-images.githubusercontent.com/49038828/139749557-516a4a96-7a0e-42b7-a4b0-bbbb4eb15bdd.png)

Pero si le hago ctrl + click al driver me lleva a una clase: 

![image](https://user-images.githubusercontent.com/49038828/139749639-067c9b6e-29d8-4edd-b5ef-532d0b075fdb.png)

Pero despues al correr el Server me tira el primer error, como si no lo tomara.

**Comment by gastonprieto** [comment, marked as answer]
hola @jcastagno99 

El archivo persistence.xml para que sea usado por los test, ponelo en src/test/resources/META-INF, me suean que no te lo detecta porque esta en el main y la dependencia es de scope=test

**Reply by jcastagno99** [reply]
Buenas, el persistence.xml lo puse en main/... porque antes lo tenia en src/test/resources/META-INF y me tiraba este error 

![image](https://user-images.githubusercontent.com/49038828/140311086-4265d981-903c-4e29-b47a-266fb4ace944.png)

lo de la version deprecada no creo que sea porque maven no me toma ninguna otra version superior a 2.4.0

![image](https://user-images.githubusercontent.com/49038828/140311482-fed8c1fc-bb6a-4a72-9698-8b0cad4dc992.png)

aunque acá me diga que la version mas reciente es la 2.6.1 http://hsqldb.org/

Puede ser que HSQLDB solo funciona dentro de un test? porque lo que estaba haciendo hasta ahora es hacerle Run a Servidor, creo que en el vídeo de la clase de Spark no hay test como tal pero no usa la base de datos usa una lista.

**Reply by gastonprieto** [reply]
Es que tenes que tener 2 persistence.xml (1 para test y otro para producción) normalmente hsqldb para test y mysql/postgresql/sql-server/etc para producción, si queres usar hsqldb para producción (main) tenes que sacarle el scope test del pom, el scope test dice que esa dependencia esta solo para test.

Para saber igual bien los errores usa `mvn test` o algun comando que ejecute la tarea, pero no directamente el IDE, porque a veces el error es por un plugin o por algun chequeo del ide que normalmente no lo daria por linea de comandos.

**Reply by jcastagno99** [reply]
Habia intentado con tener el persistence.xml tanto en main como en test, pero tampoco me tomó HSQLDB en main, ni sacando sl scope de test, al final simplemente cree un test que llame al metodo main de Server y todo funciona correctamente, así que voy a continuar de esa forma, gracias!
